src/main/rx/nrf24_v202.c

```diff
@@ -0,0 +1,295 @@
+    for (int i = 0; i < V2X2_NFREQCHANNELS; ++i) {
+        rf_channels[i] = v2x2_freq_hopping[0][i];
+    }
+    rx_timeout = 1000L;
+}
+
+static void switch_channel(void)
+{
+    NRF24L01_WriteReg(NRF24L01_05_RF_CH, rf_channels[rf_ch_num]);
+    if (++rf_ch_num >= V2X2_NFREQCHANNELS) rf_ch_num = 0;
+}
+
+STATIC_UNIT_TESTED void initV202rx(nrf24_protocol_t protocol)
+{
+    NRF24L01_Reset();
+    NRF24L01_Initialize(BV(NRF24L01_00_CONFIG_EN_CRC) | BV(NRF24L01_00_CONFIG_CRCO));
+
+    // 2-bytes CRC, radio off
+    uint8_t config = BV(NRF24L01_00_CONFIG_EN_CRC) | BV(NRF24L01_00_CONFIG_CRCO) | BV(NRF24L01_00_CONFIG_PRIM_RX);
+
+    NRF24L01_WriteReg(NRF24L01_00_CONFIG, config);
+    NRF24L01_WriteReg(NRF24L01_01_EN_AA, 0x00);      // No Auto Acknowledgment
+    NRF24L01_WriteReg(NRF24L01_02_EN_RXADDR, 0x01);  // Enable data pipe 0
+    NRF24L01_WriteReg(NRF24L01_03_SETUP_AW, 0x03);   // 5-byte RX/TX address
+    NRF24L01_WriteReg(NRF24L01_04_SETUP_RETR, 0xFF); // 4ms retransmit t/o, 15 tries
+    if (protocol == NRF24RX_V202_256K) {
+        NRF24L01_SetBitrate(NRF24L01_BR_250K);
+    } else {
+        NRF24L01_SetBitrate(NRF24L01_BR_1M);
+    }
+    NRF24L01_SetPower(3);
+    NRF24L01_WriteReg(NRF24L01_07_STATUS, 0x70);     // Clear data ready, data sent, and retransmit
+    NRF24L01_WriteReg(NRF24L01_11_RX_PW_P0, V2X2_PAYLOAD_SIZE);  // bytes of data payload for pipe 0
+    NRF24L01_WriteReg(NRF24L01_17_FIFO_STATUS, 0x00); // Just in case, no real bits to write here
+    uint8_t rx_tx_addr[] = {0x66, 0x88, 0x68, 0x68, 0x68};
+    NRF24L01_WriteRegisterMulti(NRF24L01_0A_RX_ADDR_P0, rx_tx_addr, 5);
+    NRF24L01_WriteRegisterMulti(NRF24L01_10_TX_ADDR, rx_tx_addr, 5);
+
+    delay(50);
+
+    NRF24L01_FlushTx();
+    NRF24L01_FlushRx();
+
+    rf_ch_num = 0;
+
+    // Turn radio power on
+    config |= BV(NRF24L01_00_CONFIG_PWR_UP);
+    NRF24L01_WriteReg(NRF24L01_00_CONFIG, config);
+    NRF24L01_SetTxRxMode(RX_EN);
+
+    bind_phase = PHASE_NOT_BOUND;
+    prepare_to_bind();
+
+    switch_channel();
+}
+
+static void set_bound()
+{
+    for (int i = 0; i < V2X2_NFREQCHANNELS; ++i) {
+        rf_channels[i] = freqhopping[i];
+    }
+    rx_timeout = 1000L; // find the channel as fast as possible
+}
+
+static void decode_bind_packet(uint8_t *packet)
+{
+    if ((packet[14] & V2X2_FLAG_BIND) == V2X2_FLAG_BIND) {
+        // Fill out usersettings with bound protocol parameters
+        v2x2_set_tx_id(&packet[7]);
+        // Read usersettings into current values
+        bind_phase = PHASE_BOUND;
+        set_bound();
+    }
+}
+
+// Returns whether the data was successfully decoded
+static nrf24_received_t decode_packet(uint8_t *packet)
+{
+#ifdef DEBUG_V202
+    packet[0] = 230; // throttle = 1900
+    packet[1] = 101; // yaw = 1100
+    packet[2] = 140; // pitch = 1550
+    packet[3] = 25;  // roll = 1400
+#else
+    if(bind_phase != PHASE_BOUND) {
+        decode_bind_packet(packet);
+        return NRF24_RECEIVED_BIND;
```

Comment:
no test is done if it was actually bind packet ... (but that should have no influence on code)